sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt

```diff
@@ -0,0 +1,64 @@
+package io.sentry.android.navigation
+
+import android.os.Bundle
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SentryLevel.INFO
+import io.sentry.TypeCheckHint
+import java.lang.ref.WeakReference
+
+class SentryNavigationListener @JvmOverloads constructor(
+    private val hub: IHub = HubAdapter.getInstance()
+) : NavController.OnDestinationChangedListener {
+
+    private var previousDestinationRef: WeakReference<NavDestination>? = null
+    private var previousArgs: Bundle? = null
+
+    override fun onDestinationChanged(
+        controller: NavController,
+        destination: NavDestination,
+        arguments: Bundle?
+    ) {
+        addBreadcrumb(destination, arguments)
+        previousDestinationRef = WeakReference(destination)
+        previousArgs = arguments
+    }
+
+    private fun addBreadcrumb(destination: NavDestination, arguments: Bundle?) {
+        val breadcrumb = Breadcrumb().apply {
+            type = "navigation"
+            category = "navigation"
+
+            val from = previousDestinationRef?.get()?.route
+            from?.let { data["from"] = it }
+            previousArgs?.let { args ->
+                val fromArguments = args.keySet().filter {
+                    it != NavController.KEY_DEEP_LINK_INTENT // there's a lot of unrelated stuff
+                }.associateWith { args[it] }
+                if (fromArguments.isNotEmpty()) {
+                    data["from_arguments"] = fromArguments
+                }
+            }
+
+            val to = destination.route
+            to?.let { data["to"] = it }
+            arguments?.let { args ->
+                val toArguments = args.keySet().filter {
+                    it != NavController.KEY_DEEP_LINK_INTENT // there's a lot of unrelated stuff
```

Comment:
`m`: I think we miss a test case for this edge case here.

Reply:
that's right, but I think it's not crucial for the business logic - if this condition fails and we send this data, it won't hurt the user, right? They will just see some irrelevant stuff (or maybe it's actually relevant for them - we don't know, it was just an assumption for the most users). Though I can still add the test if you want me to.

Reply:
I would remove any code if it's not worth a test. It's up to you if you want to add a test.